src/iocp/mod.rs

```diff
@@ -780,12 +877,16 @@ impl PacketUnwrapped {
 
         // Check the current status.
         match socket.status {
-            SocketStatus::Polling { readable, writable } => {
+            SocketStatus::Polling { flags } => {
                 // If we need to poll for events aside from what we are currently polling, we need
                 // to update the packet. Cancel the ongoing poll.
-                if (socket.interest.readable && !readable)
-                    || (socket.interest.writable && !writable)
-                {
+                let our_flags = event_to_afd_mask(
```

Comment:
it is probably a good idea to introduce a small wrapper around `event_to_afd_mask` that just takes such a `socket.interest` and does produce such a `our_flags`, as this pattern repeats several times...